### PR TITLE
feat/Dcv 1270 generalize test cases for dbt coves generate sources

### PR DESCRIPTION
**Pull request by SantiMenendez19**

### CODE REVIEW

File: tests/create_profiles.py [hunk 1]
```diff
@@ -13,35 +15,25 @@ def env_var(key):
 
 # Create profiles.yml
 
-with open("tests/profiles.yml.template", "r") as file:
+with open(os.path.join("tests", "templates", "profiles.yml.template"), "r") as file:
```
[the remark this answers is not in the document]
Comment (ssassi) [reply tests/create_profiles.py:18]: Agree, it shouldn't matter the cwd where you run tests from. This need to support running tests locally, not just on CI.

File: tests/create_profiles.py [hunk 2]
```diff
@@ -13,35 +15,25 @@ def env_var(key):
...
-    fields = {"profile": item["profile"]}
-    project_dir = item["project_dir"]
-    with open(f"tests/{project_dir}/dbt_project.yml", "w") as file:
+path_dir = os.path.join("tests", "generate_sources_cases")
+case_folders = glob(f"{path_dir}/*", recursive=True)
+
+
+for folder in case_folders:
+    with open(f"{folder}/settings.yml", "r") as f:
+        settings = yaml.load(f, Loader=yaml.FullLoader)
```
[the remark this answers is not in the document]
Comment (ssassi) [reply tests/create_profiles.py:36]: It's not a good idea to reuse code from the program on a test. I'd keep them separate.

File: tests/generate_sources_test.py [hunk 3]
```diff
@@ -0,0 +1,592 @@
...
+            ),
+        )
+    )
+
+
+@pytest.mark.parametrize("input", cases_list_generate_sources)
+def test_generate_sources(input):
+    # Generate sources command
+    command = [
+        "dbt-coves",
```
Comment (ssassi) [inline tests/generate_sources_test.py:305]: Isn't the argument `--profiles-dir` missing here to point to the test_case's folder?

File: tests/create_profiles.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (SantiMenendez19) [reply tests/create_profiles.py]: Solved with absolute paths and pathlib

**Pull request merged**